src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_PARAMETER = 10;
+    private Random random = new Random();
+
+    public Shape getRandomFigure() {
+        int figureType = random.nextInt(5);
+        ColorSupplier colorSupplier = new ColorSupplier();
+        String color = colorSupplier.getRandomColor();
+        switch (figureType) {
+            case 0:
+                return new Circle(color, random.nextInt(MAX_PARAMETER) + 1);
+            case 1:
+                return new Square(color, random.nextInt(MAX_PARAMETER) + 1);
+            case 2:
+                return new Rectangle(
+                    color, 
+                    random.nextInt(MAX_PARAMETER) + 1, 
+                    random.nextInt(MAX_PARAMETER) + 1
+                );
+            case 3:
+                return new IsoscelesTrapezoid(
+                    color, 
+                    random.nextInt(MAX_PARAMETER) + 1, 
+                    random.nextInt(MAX_PARAMETER) + 1, 
+                    random.nextInt(MAX_PARAMETER) + 1
```

Comment:
`random.nextInt(MAX_PARAMETER) + 1` repeats, create method for that and reuse it